### PR TITLE
Server code refactoring and minor docs updates

### PR DESCRIPTION
**Pull request by rustworthy**
## Test Plan

- Removed duplicated globals patching in dev more;
- sentry/remix can now handle `getBuild` [correctly](https://github.com/getsentry/sentry-javascript/compare/7.105.0...7.107.0#diff-73298ca0f16878e3602c16356e48fdd7d13fc660510f2475300138185eb0d712R58-R86)
- Updated docs on monitoring.

## Checklist

- [ ] Tests updated N/A
- [x] Docs updated

### CODE REVIEW

File: server/index.ts [hunk 1]
```diff
@@ -212,41 +213,36 @@ app.all(
...
-const portToUse = await getPort({
+const portToUser = await getPort({
 	port: portNumbers(desiredPort, desiredPort + 100),
 })
 
-const server = app.listen(portToUse, () => {
+const server = app.listen(portToUser, () => {
 	const addy = server.address()
-	const portUsed =
-		desiredPort === portToUse
```
Comment (rustworthy) [inline server/index.ts:228]: A tine [chance](https://github.com/sindresorhus/get-port?tab=readme-ov-file#beware), but still it makes sense to only trust the OS's final decision ;)

File: server/index.ts [hunk 2]
```diff
@@ -108,6 +107,8 @@ app.use((_, res, next) => {
 
 app.use(
 	helmet({
+		// https://github.com/helmetjs/helmet?tab=readme-ov-file#x-powered-by
```
Comment (kentcdodds) [inline server/index.ts:110]: We don't need to include links for every single thing here. People can look it up if they want.

```suggestion
```

File: server/index.ts [hunk 3]
```diff
@@ -212,41 +213,36 @@ app.all(
...
 		mode: MODE,
-		// @sentry/remix needs to be updated to handle the function signature
-		build: MODE === 'production' ? await getBuild() : getBuild,
+		build: getBuild,
 	}),
 )
 
 const desiredPort = Number(process.env.PORT || 3000)
-const portToUse = await getPort({
+const portToUser = await getPort({
```
Comment (kentcdodds) [inline server/index.ts:221]: Looks like a typo?

```suggestion
const portToUse = await getPort({
```

File: server/index.ts [hunk 4]
```diff
@@ -212,41 +213,36 @@ app.all(
...
 	port: portNumbers(desiredPort, desiredPort + 100),
 })
 
-const server = app.listen(portToUse, () => {
+const server = app.listen(portToUser, () => {
```
Comment (kentcdodds) [inline server/index.ts:225]: Typo?
```suggestion
const server = app.listen(portToUse, () => {
```

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Thanks! Just a few suggestions.

File: server/index.ts (hunk 3 above)
Comment (rustworthy) [reply server/index.ts]: It is

**Review by kentcdodds — APPROVED** [review]
Thanks!

**Pull request merged**